Chore: Cleanup and linting per pre-commit hooks

tykeal opened the pull request:
# Summary

* Cleans up _most_ of the issues that pre-commit hooks found.
* Adds in a .pre-commit-config.yaml file that can be used in conjuction with pre-commit (see pre-commit.com) and the pre-commit GitHub Action integration (see pre-commit.ci)
* Many of the pre-commit hooks were disabled as they were going to require significant changes to the codebase.

## Type of change
<!--
  What type of change does your PR introduce?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code or addition of tests

tykeal commented:
@firstof9 I'm just poking the code some to try and help cleanup where I can. I'm aware that the tests are still failing @Snuffy2 mentioned that they hadn't gotten the tests to fully pass yet either :disappointed: 

This was mostly a first step in my seeing about proposing some additional changes to the beta for quality of life and my editor was just screaming at me about all the various linting issues :rofl: so I figured I would raise the change.

I also noticed several yaml in `assets/additional_yaml` that AFAIK serve no purpose. Especially since all ZWave in HA is ZWave-JS now. I was tempted to completely remove them, but I don't know the reasoning behind them. I just dealt with the linting that was popping up from pre-commit related to them

firstof9 commented:
I fixed the tests, give me a few to get those merged in, then you can rebase this and see how everything goes.

tykeal commented:
Works for me. I would like to see the tests all working again before I start trying to add in some features, such as making all the various advanced features optional. I mean, I need some of the advanced feature, especially since I was the one that got one of them in there, but not everyone does and one of my locks ends up with over 1000 entities because of them. If I can make each of the types of advanced features optional, we'll take care of one thing I remember people asking for, making it optionally slimmer!

firstof9 commented:
I've merged the changes.

tykeal commented:
Ok, I'll pull them in and rebase off of them for my PR.

tykeal committed:
Chore: Cleanup and linting per pre-commit hooks

* Cleans up _most_ of the issues that pre-commit hooks found.
* Adds in a .pre-commit-config.yaml file that can be used in conjuction
  with pre-commit (see pre-commit.com) and the pre-commit GitHub Action
  integration (see pre-commit.ci)
* Many of the pre-commit hooks were disabled as they were going to require
  significant changes to the codebase.

Signed-off-by: Andrew Grimberg <tykeal@bardicgrove.org>

tykeal commented:
@firstof9 I've updated the linting. Looks like we're just failing on coverage? I noticed that your PR is also failing on coverage percentage.

firstof9 commented:
Yes 59% seems to be as close as I can get currently.

firstof9 commented on custom_components/keymaster/config_flow.py:
```diff
@@ -297,7 +297,7 @@ async def _start_config_flow(
             if step_id == "user" or not entry_id:
                 return cls.async_create_entry(title=title, data=user_input)
-            cls.hass.config_entries.async_update_entry(cls._entry, data=user_input)  # type: ignore[arg-type]
+            cls.hass.config_entries.async_update_entry(cls._entry, data=user_input)
```
```suggestion
            cls.hass.config_entries.async_update_entry(cls._entry, data=user_input)  # type: ignore[arg-type]
```

firstof9 replied:
Needed for mypy

tykeal replied:
mypy run under pre-commit throws the following if I have that in there:

```
custom_components/keymaster/config_flow.py:300: error: Unused "type: ignore" comment  [unused-ignore]
```

I've instead modified the code itself. Hopefully the change I made is actually fully correct. When I ran the tests without the ignore, mypy complained of the value being `Any | None` when the function expects `Any`

firstof9 replied:
Before your changes mypy was getting hung up there. This seems to work.

firstof9 commented:
@tykeal All set?

tykeal commented on custom_components/keymaster/config_flow.py:
```diff
@@ -297,9 +297,10 @@ async def _start_config_flow(
-            cls.hass.config_entries.async_update_entry(cls._entry, data=user_input)  # type: ignore[arg-type]
-            await cls.hass.config_entries.async_reload(entry_id)
-            return cls.async_abort(reason="reconfigure_successful")
+            if cls._entry is not None:
+                cls.hass.config_entries.async_update_entry(cls._entry, data=user_input)
+                await cls.hass.config_entries.async_reload(entry_id)
+                return cls.async_abort(reason="reconfigure_successful")
```
@firstof9 if you're fine with this change then yes, I'm good for this to merge. AFAIK it _should_ do the correct thing given that we shouldn't end up in a situation where `cls._entry` is `None` here

firstof9 replied:
Yes, the test would have picked up an error here if there were issues.
